File: source/livana-fe/src/types/request/updateExperienceListingRequest.ts
import type { ImageOrderDto } from "./createExperienceListingRequest";

export interface UpdateExperienceListingRequest {
  title: string;
  price: number;
  description: string;
  capacity: number;
  address: string;
  latitude: number;
  longitude: number;
  experienceCategoryId: string;
  images: ImageOrderDto[];
}

File: source/livana-be/src/main/java/octguy/livanabe/dto/request/UpdateExperienceListingRequest.java
package octguy.livanabe.dto.request;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;
import octguy.livanabe.dto.dto.ImageOrderDto;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

@Data
public class UpdateExperienceListingRequest {

    @NotBlank(message = "Title is required")
    private String title;

    @Min(value = 1, message = "Price must be greater than 0")
    private BigDecimal price;

    private String description;

    private int capacity;

    private String address;

    private Double latitude;

    private Double longitude;

    private UUID experienceCategoryId;

    private List<ImageOrderDto> images;
}
